StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1008OpeningParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -94,66 +94,70 @@ private void HandleOpenParenToken(SyntaxTreeAnalysisContext context, SyntaxToken
-                    break;
+                    case SyntaxKind.AwaitKeyword:
+                    case SyntaxKind.CaseKeyword:
+                    case SyntaxKind.CatchKeyword:
+                    case SyntaxKind.FixedKeyword:
+                    case SyntaxKind.ForKeyword:
+                    case SyntaxKind.ForEachKeyword:
+                    //case SyntaxKind.FromKeyword: // ?
+                    case SyntaxKind.GroupKeyword: // ?
+                    case SyntaxKind.IfKeyword:
+                    //case SyntaxKind.IntoKeyword: // ?
+                    //case SyntaxKind.JoinKeyword: // ?
+                    //case SyntaxKind.LetKeyword: // ?
+                    case SyntaxKind.LockKeyword:
+                    case SyntaxKind.OrderByKeyword: // ?
+                    case SyntaxKind.ReturnKeyword:
+                    case SyntaxKind.SelectKeyword: // ?
+                                                   //case SyntaxKind.StackAllocKeyword: // ?
+                    case SyntaxKind.SwitchKeyword:
+                    case SyntaxKind.UsingKeyword:
+                    case SyntaxKind.WhereKeyword: // ?
+                    case SyntaxKind.WhileKeyword:
+                        allowLeadingNoSpace = false;
+                        allowLeadingSpace = true;
+                        // allow these to be reported as SA1000
+                        reportPrecedingError = false;
+                        break;
 
-                case SyntaxKind.EqualsGreaterThanToken: // lambda containing a cast
-                case SyntaxKind.CommaToken:
-                    allowLeadingNoSpace = false;
-                    allowLeadingSpace = true;
-                    break;
+                    case SyntaxKind.CheckedKeyword:
+                    case SyntaxKind.DefaultKeyword:
+                    case SyntaxKind.SizeOfKeyword:
+                    case SyntaxKind.TypeOfKeyword:
+                    case SyntaxKind.UncheckedKeyword:
+                        allowLeadingNoSpace = true;
+                        allowLeadingSpace = false;
+                        // allow these to be reported as SA1000
+                        reportPrecedingError = false;
+                        break;
 
-                default:
-                    if (precedingToken.Parent is BinaryExpressionSyntax
-                        || precedingToken.Parent is AssignmentExpressionSyntax
-                        || precedingToken.Parent is ConditionalExpressionSyntax
-                        || precedingToken.Parent is EqualsValueClauseSyntax)
-                    {
+                    case SyntaxKind.EqualsGreaterThanToken: // lambda containing a cast
+                    case SyntaxKind.CommaToken:
                         allowLeadingNoSpace = false;
                         allowLeadingSpace = true;
                         break;
-                    }
-                    else
-                    {
+                    case SyntaxKind.NewKeyword:
```

Comment:
This entire case statement case be removed by adding `case SyntaxKind.NewKeyword:` after one of the following (and removing the contents of this case), depending on the desired behavior. See the comment on the pull request for more information.

* After `case SyntaxKind.LockKeyword:`, if the desired spacing is `new ()`
* After `case SyntaxKind.DefaultKeyword:` if the desired spacing is `new()`